### PR TITLE
BUG: Error upon Series.Groupby.nunique with empty Series (#12553)

### LINKED ISSUE #12553: SeriesGroupby.nunique raises an IndexError on empty Series
#### Code Sample, a copy-pastable example if possible

``` python
In [18]: b = pandas.Series()

In [19]: g = b.groupby(level = 0)

In [20]: g.nunique()
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
<ipython-input-20-fbbfc3108eac> in <module>()
----> 1 g.nunique()

/usr/local/lib/python2.7/dist-packages/pandas/core/groupby.pyc in nunique(self, dropna)
   2693 
   2694         out = np.add.reduceat(inc, idx).astype('int64', copy=False)
-> 2695         return Series(out if ids[0] != -1 else out[1:],
   2696                       index=self.grouper.result_index,
   2697                       name=self.name)

IndexError: index 0 is out of bounds for axis 0 with size 0
```
#### Expected Output

``` python
0
```
#### output of `pd.show_versions()`

```
INSTALLED VERSIONS
------------------
commit: None
python: 2.7.6.final.0
python-bits: 64
OS: Linux
OS-release: 3.13.0-79-generic
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8

pandas: 0.17.1
nose: 1.3.1
pip: 8.0.2
setuptools: 20.1.1
Cython: None
numpy: 1.10.4
scipy: 0.13.3
statsmodels: 0.5.0
IPython: 4.1.1
sphinx: None
patsy: 0.2.1
dateutil: 2.4.2
pytz: 2015.7
blosc: None
bottleneck: None
tables: 3.1.1
numexpr: 2.2.2
matplotlib: 1.3.1
openpyxl: 1.7.0
xlrd: 0.9.2
xlwt: 0.7.5
xlsxwriter: None
lxml: None
bs4: 4.2.1
html5lib: 0.999
httplib2: 0.8
apiclient: None
sqlalchemy: None
pymysql: None
psycopg2: None
Jinja2: None
```

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] closes #12553
 - [x] tests added / passed
 - [x] passes ``git diff upstream/master | flake8 --diff``
 - [x] whatsnew entry

Similar to #12560, but it has appeared to stall. Wanted to push it along.

### CODE REVIEW

**Comment by MaximilianR** [comment]
I think nunique was one of a few cases I hadn't completed when trying to solve some of these groupby + empty series. Generally I had tried to remove empty as a special case. But if that's not possible, sobeit.

Regardless we should use `shallow_copy` rather than `Series` here

File: pandas/core/groupby.py [hunk 1]
```diff
@@ -2898,6 +2898,10 @@ def true_and_notnull(x, *args, **kwargs):
     def nunique(self, dropna=True):
         """ Returns number of unique elements in the group """
         ids, _, _ = self.grouper.group_info
+
+        if len(ids) == 0:
+            return Series(index=self.dtype.index, name=self.name)
```
Comment (jreback) [inline pandas/core/groupby.py]: ``index=self.obj.index``

File: pandas/core/groupby.py [hunk 2]
```diff
@@ -2898,6 +2898,10 @@ def true_and_notnull(x, *args, **kwargs):
...
+
         val = self.obj.get_values()
```
Comment (jreback) [inline pandas/core/groupby.py]: actualy I'd rather  you fix like this:

```
diff --git a/pandas/core/groupby.py b/pandas/core/groupby.py
index afddb86..f720a62 100644
--- a/pandas/core/groupby.py
+++ b/pandas/core/groupby.py
@@ -2928,7 +2928,10 @@ class SeriesGroupBy(GroupBy):
             inc[idx] = 1
 
         out = np.add.reduceat(inc, idx).astype('int64', copy=False)
-        res = out if ids[0] != -1 else out[1:]
+        if len(ids):
+            res = out if ids[0] != -1 else out[1:]
+        else:
+            res = out[1:]
         ri = self.grouper.result_index
 
         # we might have duplications among the bins
```

**Comment by mroeschke** [comment]
Thanks @jreback, moved the `len` check down the stack. Just waiting for Travis to finish the build.

**Comment by jreback** [comment]
thanks @mroeschke

**Pull request closed without merging**